src/main/java/com/github/shyiko/mysql/binlog/event/deserialization/BinaryLogEventDataReader.java

```diff
@@ -0,0 +1,313 @@
+     * Reads {@code length} bytes from the current position in the buffer and interprets it as little-endian int
+     * returning the value.
+     */
+    public int readInteger(int length) {
+        // Specific getXXX methods are usually faster due to internal use of Unsafe.
+        switch (length) {
+            case 4:
+                return buffer.getInt();
+            case 2:
+                return buffer.getShort() & 0xFFFF;
+            case 1:
+                return buffer.get() & 0xFF;
+        }
+
+        int result = 0;
+        for (int i = 0; i < length; i++) {
+            result |= (buffer.get() & 0xFF) << (i << 3);
+        }
+        return result;
+    }
+
+    /**
+     * Reads {@code length} bytes from the current position in the buffer and interprets it as big-endian int
+     * returning the value.
+     */
+    public int readIntegerBE(int length) {
+        switch (length) {
+            case 4:
+                return Integer.reverseBytes(buffer.getInt());
+            case 2:
+                return Short.reverseBytes(buffer.getShort()) & 0xFFFF;
+            case 1:
+                return buffer.get() & 0xFF;
+        }
+
+        int result = 0;
+        for (int i = 0; i < length; i++) {
+            result <<= 8;
+            result |= buffer.get() & 0xFF;
+        }
+        return result;
+    }
+
+    public int readUnsignedByte() {
+        return buffer.get() & 0xFF;
+    }
+
+    /**
+     * Returns the value of the byte at the current position in the buffer or -1 if there are no bytes left.
+     */
+    public int peekUnsignedByte() {
+        if (!buffer.hasRemaining()) {
+            return -1;
+        }
+        return buffer.get(buffer.position()) & 0xFF;
+    }
+
+    public byte readByte() {
+        return buffer.get();
+    }
+
+    /**
+     * Reads {@code length} bytes from the current position in the buffer returning them as byte array.
+     */
+    public byte[] readBytes(int length) {
+        byte[] bytes = new byte[length];
+        buffer.get(bytes, 0, bytes.length);
+        return bytes;
+    }
+
+    public void skip(int length) {
+        buffer.position(buffer.position() + length);
+    }
+
+    /**
+     * Reads packed long value from the current position in the buffer.
+     * <p>
+     * Format (first-byte-based):<br>
+     * 0-250 - The first byte is the number (in the range 0-250). No additional bytes are used.<br>
+     * 251 - SQL NULL value<br>
+     * 252 - Two more bytes are used. The number is in the range 251-0xffff.<br>
+     * 253 - Three more bytes are used. The number is in the range 0xffff-0xffffff.<br>
+     * 254 - Eight more bytes are used. The number is in the range 0xffffff-0xffffffffffffffff.
+     * @return long or null
+     */
+    public long readPackedLong() {
+        int b = (buffer.get() & 0xFF);
+        if (b < 251) {
+            return b;
+        } else if (b == 251) {
+            throw new RuntimeException("Unexpected NULL value");
```

Comment:
Should we keep this as an `IOException` to preserve the same behavior as `ByteArrayInputStream#readPackedLong`?

Reply:
That's a good question. I preserved `IOException` in some methods to keep the contract similar to the one `ByteArrayInputStream` provides to minimize changes. I can't recall why I decided not to do the same in this method. I agree that it might be better to use IOException here for now as well.

Reply:
Fixed